components/datetime/src/format/datetime.rs

```diff
@@ -547,126 +451,136 @@ where
                             h
                         }
                     }
-                });
-
-            let () = match value {
-                Ok(value) => {
-                    r = r.and(try_write_number(
-                        w,
-                        fixed_decimal_format,
-                        FixedDecimal::from(value),
-                        field.length,
-                    )?)
-                }
-                Err(e) => {
-                    r = r.and(Err(e));
-                    w.with_part(Part::ERROR, |w| "<ICU4X ERROR>".write_to(w))?
-                }
-            };
+                };
+                try_write_number(w, fdf, h.into(), l)?
+            } else {
+                w.with_part(Part::ERROR, |w| {
+                    try_write_number(w, fdf, 0.into(), l).map(|_| ())
+                })?;
```

Comment:
Issue: This should use `write_value_missing`. I don't like falling back to `0` for anything except _maybe_ fractional seconds.